src/services/SendEmailProcessor.ts

```diff
@@ -31,7 +31,7 @@ export class SendEmailProcessor {
 		this.validationHelper = new ValidationHelper();
 		this.metrics = metrics;
 		this.govNotifyService = SendEmailService.getInstance(this.logger, GOVUKNOTIFY_API_KEY);
-		this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.SEND_EMAIL_PROCESSOR_SERVICE);
+		this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.GOV_NOTIFY_SERVICE);
 		this.iprService = IPRService.getInstance(this.environmentVariables.sessionEventsTable(), this.logger, createDynamoDbClient());
```

Comment:
Sorry why are we using environment variables that aren't for this service? Is this going to mean that is we need to add any that are specifically for `SendEmailProcessor` in the future we're going to have to add them to `GOV_NOTIFY_SERVICE` instead?

Reply:
SendEmailProcessor is only referenced by GovNotify service and hence making use of this service enum. Ideally best approach would be to only pass on the table name from the handler/caller to initiate the ipr service. We have environment variables check in place per each function(service) as we set the same in template.yaml.

Reply:
Refactored the code to pass the table name from the handler to avoid any confusions. Tested e2e and works fine.